dev/io.openliberty.microprofile.openapi.2.0.internal_fat/fat/src/io/openliberty/microprofile/openapi20/fat/deployments/MergeServerXMLTest.java

```diff
@@ -0,0 +1,258 @@
+                                          .addAsModules(war1, war2, war3);
+
+        deployAppToApps(ear, appName, "ear");
+
+        String doc = OpenAPIConnection.openAPIDocsConnection(server, false).download();
+        JsonNode openapiNode = OpenAPITestUtil.readYamlTree(doc);
+        OpenAPITestUtil.checkPaths(openapiNode, 2, "/test1/test", "/test2/test");
+        OpenAPITestUtil.checkInfo(openapiNode, "Generated API", "1.0");
+        assertServerContextRoot(openapiNode, null);
+    }
+
+    @Test
+    public void testMultiModuleEarComplexServerXML() throws Exception {
+
+        MpOpenAPIElement.MpOpenAPIElementBuilder.cloneBuilderFromServerResetAppsAndModules(server)
+                                                .addIncludedApplicaiton("testEar")
+                                                .addExcludedModule("testEar/test3")
+                                                .addExcludedModule("testEar/test4")
+                                                .buildAndPushToServer();
+
+        WebArchive war1 = ShrinkWrap.create(WebArchive.class, "test1.war")
+                                    .addClasses(DeploymentTestApp.class, DeploymentTestResource.class);
+
+        WebArchive war2 = ShrinkWrap.create(WebArchive.class, "test2.war")
+                                    .addClasses(DeploymentTestApp.class, DeploymentTestResource.class);
+
+        WebArchive war3 = ShrinkWrap.create(WebArchive.class, "test3.war")
+                                    .addClasses(DeploymentTestApp.class, DeploymentTestResource.class);
+
+        WebArchive war4 = ShrinkWrap.create(WebArchive.class, "test4.war")
+                                    .addClasses(DeploymentTestApp.class, DeploymentTestResource.class);
+
+        EnterpriseArchive ear = ShrinkWrap.create(EnterpriseArchive.class, "testEar.ear")
+                                          .addAsModules(war1, war2, war3, war4);
+
+        deployApp(ear);
+
+        logOnServer("/test1/log", "testMultiModuleEarComplexServerXML");
+
+        String doc = OpenAPIConnection.openAPIDocsConnection(server, false).download();
+        JsonNode openapiNode = OpenAPITestUtil.readYamlTree(doc);
+        OpenAPITestUtil.checkPaths(openapiNode, 2, "/test1/test", "/test2/test");
+        OpenAPITestUtil.checkInfo(openapiNode, "Generated API", "1.0");
+        assertServerContextRoot(openapiNode, null);
+    }
+
+    @Test
+    public void testMPConfigIgnoredInServerXMLMode() throws Exception {
+
+        MpOpenAPIElement.MpOpenAPIElementBuilder.cloneBuilderFromServerResetAppsAndModules(server)
+                                                .addIncludedApplicaiton("all")
+                                                .buildAndPushToServer();
+
+        //The combo of all and an exclude should be all except testEar/test2, but since this is set via
+        //mpConfig this should be ignored.
+        setMergeConfig(null, "testEar/test2", null);
```

Comment:
This doesn't test what it's meant to because we can't actually update the system properties here while the server is running.

Noted this as something to fix in #29874